### ISSUE TITLE
C++ plugin cannot handle large input

### ISSUE THREAD

**Issue by gaow** [stephenslab/mashr#70]
Separate from #69 : when input to MASH is large,
```
- Computing 39972291 x 111 likelihood matrix.

error: Mat::init(): requested size is too large
Error in calc_lik_rcpp(t(data$Bhat), t(data$Shat), data$V, matrix(0, 0,  :
  Mat::init(): requested size is too large
```

Apparently [the problem comes from `armadillo`](https://stackoverflow.com/questions/23573667/armadillo-requested-size-is-too-large) and the solution seems to involve changing compile flags.

**Comment by gaow** [comment]
@pcarbo yes I saw that, but did not think too much when our unit tests still pass. I agree this may be a valid concern as we don't know why armadillo chose to do that (a quick search did not answer it). Maybe let's wait for @jlboat to report back -- if it works I'll move it to another branch and ask people to use that in-house to test; if it does not work I'll revert?

**Comment by jlboat** [comment]
Error in calc_lik_rcpp(t(data$Bhat), t(data$Shat), data$V, matrix(0, 0,  : 
  std::bad_alloc
Calls: mash ... calc_relative_lik_matrix -> calc_lik_matrix -> calc_lik_rcpp
Timing stopped at: 9.963 1.71 11.71
Execution halted

**Comment by jlboat** [comment]
And just to clarify, I believe this run had plenty of memory allocated:

JobName    Mem  CPUs   User         Walltime               %CPU    %MEM   
mashr_old   372gb   16  jboatw2    04:58/160:00 ( 3%)    97%      21%   
mashr_new  372gb  16  jboatw2    02:31/160:00 ( 2%)    97%      27% 

Running mashr with ARMA_64BIT_WORD consistently ran with about 6% higher memory and did not appear to exceed the 372 Gb max.
